sql/sql_parse.cc

```diff
@@ -5535,8 +5544,7 @@ int mysql_execute_command(THD *thd, bool first_level) {
         WSREP_SYNC_WAIT(thd, WSREP_SYNC_WAIT_BEFORE_SHOW);
       }
 
-      if (lex->sql_command == SQLCOM_ALTER_USER_DEFAULT_ROLE ||
```

Comment:
It is not necessary. TOI was started here and then `mysql_alter_or_clear_default_roles()` was called where TOI start attempt was made again. That 2nd attempt was skipped because already in TOI.

Reply:
Okay. Got it. Thanks for the explanation.